server/src/main/java/org/opensearch/cluster/metadata/ResourceLimitGroup.java

```diff
@@ -0,0 +1,288 @@
+    public static final ParseField MODE_FIELD = new ParseField("mode");
+
+    @SuppressWarnings("unchecked")
+    private static final ConstructingObjectParser<ResourceLimitGroup, Void> PARSER = new ConstructingObjectParser<>(
+        "ResourceLimitGroupParser",
+        args -> new ResourceLimitGroup((String) args[0], (List<ResourceLimit>) args[1], (String) args[2])
+    );
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), NAME_FIELD);
+        PARSER.declareObjectArray(
+            ConstructingObjectParser.constructorArg(),
+            (p, c) -> ResourceLimit.fromXContent(p),
+            RESOURCE_LIMITS_FIELD
+        );
+        PARSER.declareString(ConstructingObjectParser.constructorArg(), MODE_FIELD);
+    }
+
+    public ResourceLimitGroup(final String name, final List<ResourceLimit> resourceLimits, final String modeName) {
+        Objects.requireNonNull(name, "ResourceLimitGroup.name can't be null");
+        Objects.requireNonNull(resourceLimits, "ResourceLimitGroup.resourceLimits can't be null");
+
+        if (name.length() > MAX_CHARS_ALLOWED_IN_NAME) {
+            throw new IllegalArgumentException("ResourceLimitGroup.name shouldn't be more than 50 chars long");
+        }
+
+        if (resourceLimits.isEmpty()) {
+            throw new IllegalArgumentException("ResourceLimitGroup.resourceLimits should at least have 1 resource limit");
+        }
+
+        this.name = name;
+        this.resourceLimits = resourceLimits;
+        this.mode = ResourceLimitGroupMode.fromName(modeName);
+    }
+
+    public ResourceLimitGroup(StreamInput in) throws IOException {
+        this(in.readString(), in.readList(ResourceLimit::new), in.readString());
+    }
+
+    /**
+     * Class to hold the system resource limits;
+     * sample Schema
+     *
+     */
+    @ExperimentalApi
+    public static class ResourceLimit implements Writeable, ToXContentObject {
+        private final String resourceName;
+        private final Double value;
+
+        static final ParseField RESOURCE_NAME_FIELD = new ParseField("resourceName");
+        static final ParseField RESOURCE_VALUE_FIELD = new ParseField("value");
+
+        public static final ConstructingObjectParser<ResourceLimit, Void> PARSER = new ConstructingObjectParser<>(
+            "ResourceLimitParser",
+            args -> new ResourceLimit((String) args[0], (Double) args[1])
+        );
+
+        static {
+            PARSER.declareString(ConstructingObjectParser.constructorArg(), RESOURCE_NAME_FIELD);
+            PARSER.declareDouble(ConstructingObjectParser.constructorArg(), RESOURCE_VALUE_FIELD);
+        }
+
+        public ResourceLimit(String resourceName, Double value) {
+            Objects.requireNonNull(resourceName, "resourceName can't be null");
+            Objects.requireNonNull(value, "resource value can't be null");
+
+            if (Double.compare(value, 1.0) > 0) {
```

Comment:
```suggestion
            if (value > 1.0) {
```
Probably this is much simpler ?